compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -558,6 +564,11 @@ impl Diagnostic {
         self
     }
 
+    pub fn set_is_lint(&mut self, is_lint_diagnostic: bool) -> &mut Self {
+        self.is_lint = is_lint_diagnostic;
+        self
+    }
```

Comment:
This is only being called to set it to `true`.

```suggestion
    pub fn set_is_lint(&mut self) -> &mut Self {
        self.is_lint = true;
        self
    }
```